src/realm/object-store/sync/app.cpp

```diff
@@ -232,6 +236,11 @@ App::App(const Config& config)
     , m_auth_route(m_app_route + auth_path)
     , m_request_timeout_ms(m_config.default_request_timeout_ms.value_or(default_timeout_ms))
 {
+#ifdef __EMSCRIPTEN__
+    if (!m_config.transport) {
+        m_config.transport = std::make_shared<_impl::EmscriptenNetworkTransport>();
+    }
+#endif
```

Comment:
Is emscripten only going to be used with the WASM SDK? If so, what would you think about moving the transport and socketprovider classes to the SDK?

Reply:
Many of our SDKs can have wasm builds - realm-js is the priority now, but realm-dotnet is sure to follow, based on requests. Dart and Kotlin can also benefit. Since the networking code is browser-specific, but multiple SDKs could target the browser I opted to have it live in Core and be the default unless an SDKs chooses to do otherwise. However, since we have little control over the browser's network stack and our customization opportunities are limited I don't know if SDKs would ever need to override the defaults, expect maybe in tests where they mock the server endpoints.

Reply:
Yes - ideally I'd like to run Core's existing test suite, but right now we can only build them, not run them. The big blocker there is that virtually all of our tests expect filesystem persistence, and a lot of them rely on being able to spawn threads, both of which are unsupported right now. But as the project matures the expectation is that we'll be running Core tests on the target.